src/DotLiquid/ExtendedFilters.cs

```diff
@@ -38,5 +40,83 @@ public static string UpcaseFirst(Context context, string input)
             var trimmed = input.TrimStart();
             return input.Substring(0, input.Length - trimmed.Length) + char.ToUpper(trimmed[0]) + trimmed.Substring(1);
         }
+
+        /// <summary>
+        /// Convert a UNIX Epoch timestamp in milliseconds into a UTC DateTimeOffset.
+        /// </summary>
+        /// <param name="context">The DotLiquid context</param>
+        /// <param name="input">string/long containing milliseconds</param>
+        public static object UnixMs(Context context, object input)
+        {
+            if ((input is double) || (input is long) || (input is ulong))
+            {
+                return CreateDateTimeOffsetFromUnixMilliseconds(Convert.ToInt64(input, context.FormatProvider));
+            }
+            else if (input is string stringInput &&
+                long.TryParse(stringInput, NumberStyles.Integer, CultureInfo.InvariantCulture, out var timestamp))
+            {
+                return CreateDateTimeOffsetFromUnixMilliseconds(timestamp);
+            }
+
+            return input;
+        }
+
+        /// <summary>
+        /// Convert a date-time input to a different timezone.
+        /// </summary>
+        /// <param name="context">The DotLiquid context</param>
+        /// <param name="input">DateTimeOffset, DateTime (unspecified, local or UTC) or a date string</param>
+        /// <param name="convertToTimezoneId">Windows timezone ID to convert to</param>
+        /// <seealso href="https://docs.microsoft.com/en-us/dotnet/api/system.timezoneinfo.findsystemtimezonebyid?view=netcore-3.1#remarks"/>
+        public static object TimeZone(Context context, object input, string convertToTimezoneId)
+        {
+            // Convert the input to a DateTimeOffset
+            if (input is DateTimeOffset dateTimeOffset ||
+                (input is string stringInput && DateTimeOffset.TryParse(stringInput, out dateTimeOffset)))
```

Comment:
I don't agree with parsing in here, created extra overhead/testing.

Reply:
The date filter returns a string, so without string parsing time_zone cannot be chained after a date filter, which limits it's usability.
How about if I changed it to use TryParseExact with a fixed format of say ISO-8601 with explicit timezone specifier.
That would reduce the overhead and simplify testing to a narrower set of input formats.

Reply:
Good idea. I'd accept it if it was limited in scope as you describe.

Reply:
Changes made and docs drafted at https://github.com/robin-parker/dotliquid/wiki/time_zone